File: U1/EXAMEN/Comercializadora_Soap_DotNet_GR01/Comercializadora_Soap_DotNet_GR01/Models/Producto.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Comercializadora_Soap_DotNet_GR01.Models
{
    [Table("Producto")]
    public class Producto
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int ProductoId { get; set; }

        [Required]
        [MaxLength(20)]
        [Index("IX_Producto_Codigo", IsUnique = true)]
        public string Codigo { get; set; }

        [Required]
        [MaxLength(100)]
        public string Nombre { get; set; }

        [MaxLength(500)]
        public string Descripcion { get; set; }

        [Required]
        [Column(TypeName = "decimal(12,2)")]
        public decimal Precio { get; set; }

        [Required]
        public int Stock { get; set; }

        [MaxLength(50)]
        public string Categoria { get; set; }

        [MaxLength(500)]
        public string ImagenUrl { get; set; }

        [Required]
        public DateTime FechaRegistro { get; set; }

        [Required]
        [MaxLength(20)]
        public string Estado { get; set; }

        // Navigation properties
        public virtual ICollection<DetalleFactura> DetallesFactura { get; set; }

        public Producto()
        {
            FechaRegistro = DateTime.Now;
            Estado = "ACTIVO";
            Stock = 0;
            DetallesFactura = new HashSet<DetalleFactura>();
        }
    }
}

File: U2/PROYECTO/Comercializadora_RestFul_Java_GR01/src/main/java/ec/edu/monster/Models/Producto.java
package ec.edu.monster.Models;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.HashSet;
import java.util.Set;

/**
 * Entidad Producto
 */
@Entity
@Table(name = "Producto")
@Data
@NoArgsConstructor
public class Producto {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "ProductoId")
    private Integer productoId;

    @Column(name = "Codigo", nullable = false, length = 20, unique = true)
    private String codigo;

    @Column(name = "Nombre", nullable = false, length = 100)
    private String nombre;

    @Column(name = "Descripcion", length = 500)
    private String descripcion;

    @Column(name = "Precio", nullable = false, precision = 12, scale = 2)
    private BigDecimal precio;

    @Column(name = "Stock", nullable = false)
    private Integer stock;

    @Column(name = "Categoria", length = 50)
    private String categoria;

    @Column(name = "ImagenUrl", length = 500)
    private String imagenUrl;

    @Column(name = "FechaRegistro", nullable = false)
    private LocalDateTime fechaRegistro;

    @Column(name = "Estado", nullable = false, length = 20)
    private String estado;

    // Relación con DetalleFactura
    @OneToMany(mappedBy = "producto", fetch = FetchType.LAZY)
    private Set<DetalleFactura> detallesFactura = new HashSet<>();

    @PrePersist
    protected void onCreate() {
        if (fechaRegistro == null) {
            fechaRegistro = LocalDateTime.now();
        }
        if (estado == null) {
            estado = "ACTIVO";
        }
        if (stock == null) {
            stock = 0;
        }
    }
}
